proposed/standard-ci-env-variables.md

```diff
@@ -0,0 +1,47 @@
+# Standardized Environment Variables for CI Services
+
+Many CI services set environment variables that can be used by developer builds. Many of the environment variables are common in scope and theme across CI services, however, the specific variable names are different. We need a set of standardized environment variables across CI services so that an application platform like .NET can rely on them, and provide a common user experience across CI services.
+
+## Context
+
+The .NET Core SDK is implementing new scenarios that require source control manager information. An initial idea was to call out to specific source control tools to get this information. This approach is problematic since it requires an implementation for each source control manager and it has the potential to be fragile and slow.
+
+It turns out that CI services all of the information that these new .NET Core scenarios need via environment variables. Also, there is a strong relationship between these scenarios and official CI-provided builds (much less on local developer builds). As a result, relying on the CI-provided environment variables is attractive.
+
+To make it possible to provide application platform provided experiences across CI services, we need a standardized set of environment variables that are supported across those same CI services. Ideally, this set of environment variables would be supported across multiple CI services and useful for multiple application environments, not just .NET.
+
+## Proposed environment variables
+
+The .NET Core SDK needs are oriented around source control. As a result, the intial list is source control oriented, but there is no affinity to source control on the general idea of standardized environment variables.
```

Comment:
I think the specification for some the proposed environment variables is too vague. The consuming application probably relies on a specific format or otherwise needs to deal with missing/unexpected info.

E.g. can `STANDARDCI_REPOSITORYCOMMITID` only be a git hash? If not, this needs to be treated as a free form string.  Same for `STANDARDCI_REPOSITORYNAME` unless we want to prescribe the org/repo format.

Can any of the variables be empty if no sensible information is available?

Reply:
@akoeplinger Good points. We should specify what we expect of the value. 

There are a few uses of the commit id we are thinking of at this moment:

1) Use it in [source link](https://github.com/dotnet/core/blob/master/Documentation/diagnostics/source_link.md) to map local paths to URLs on source server.

This is the most important usage that also implies most restrictions on the string. @richlander we need to get in touch with various source control providers to see what URLs are they using for their content servers. 

1) Use it in ```AssemblyInformationVersion``` attribute like so: 

```xml
<InformationalVersion>$(Version). Commit: $(STANDARDCI_REPOSITORYCOMMITID)</InformationalVersion>
```

This way one can identify the corresponding source commit easily just by looking at an assembly level attribute. This is just a display string with no specific format requirements.

1) Include it in a description of a NuGet package again to allow associating particular package version with a source commit. This is also a display string with no specific format requirements.